### PR TITLE
 Convert RuntimeMessageAdapter.java to Kotlin

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: wire-runtime/build.gradle [hunk 1]
```diff
@@ -53,6 +53,10 @@ task japicmp(type: JapicmpTask, dependsOn: 'jar') {
   packageExcludes = [
       'com.squareup.wire.internal',
   ]
+  classExcludes = [
+      // package protected -> public    
+      'com.squareup.wire.RuntimeMessageAdapter',
```
Comment (Egorand) [inline wire-runtime/build.gradle]: Gets accessed from `wire-gson-support` so has to be public

File: wire-runtime/src/main/java/com/squareup/wire/FieldBinding.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
  * Read, write, and describe a tag within a message. This class knows how to assign fields to a
  * builder object, and how to extract values from a message object.
  */
-final class FieldBinding<M extends Message<M, B>, B extends Message.Builder<M, B>> {
+public final class FieldBinding<M extends Message<M, B>, B extends Message.Builder<M, B>> {
```
Comment (Egorand) [inline wire-runtime/src/main/java/com/squareup/wire/FieldBinding.java]: Is exposed through `RuntimeMessageAdapter` hence should also become public

File: wire-runtime/build.gradle (hunk 1 above)
Comment (JakeWharton) [reply wire-runtime/build.gradle]: Is this still needed?

File: wire-runtime/src/main/java/com/squareup/wire/FieldBinding.java (hunk 2 above)
Comment (JakeWharton) [reply wire-runtime/src/main/java/com/squareup/wire/FieldBinding.java]: Should also be in 'internal'

Comment (Egorand) [reply wire-runtime/src/main/java/com/squareup/wire/FieldBinding.java]: That's in a different PR

File: wire-runtime/build.gradle (hunk 1 above)
Comment (Egorand) [reply wire-runtime/build.gradle]: Thought it would be since the class is missing. Japicmp doesn't complain though, so I removed it.

**Pull request merged**